### PR TITLE
DDD-17 Setup initial entities

### PR DESCRIPTION
**Pull request by KornelisMax**
Created entities and associations between them according to our entity [diagram](https://ba-jira.atlassian.net/jira/software/projects/DDD/boards/5?selectedIssue=DDD-17) (file attached in jira ticket).
In addition, created one to many relationship between user and order (we are planning to add functionality to see order history for registered user). For now, I added `CASCADE ALL` actions only to `user`, `user_info`, `address` fields, because only these fields are going to be edited together (but probably this needs more investigation). 

JIRA task: https://ba-jira.atlassian.net/jira/software/projects/DDD/boards/5?selectedIssue=DDD-17

@egisxxegis @Gintasp @Accir @Zygis0321

### CODE REVIEW

File: src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
         <jta-data-source>AppDataSource</jta-data-source>
         <properties>
             <property name="hibernate.show_sql" value="true"/>
-            <property name="hibernate.hbm2ddl.auto" value="update"/>
+            <property name="hibernate.hbm2ddl.auto" value="create"/>
```
Comment (KornelisMax) [inline src/main/resources/META-INF/persistence.xml]: After first launch don't forget to change it back to `value="update"`

File: src/main/java/lt/vu/persistence/orm/entities/Address.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    private String street;
+
+    private String nr;
+
+    @OneToOne
+    private User user;
+
+    @OneToOne
+    private UserInfo guest;
```
Comment (KornelisMax) [inline src/main/java/lt/vu/persistence/orm/entities/Address.java]: We should decide on registered/guest user naming. I think guest fits quite well here.

File: src/main/resources/META-INF/persistence.xml (hunk 1 above)
Comment (giplonis) [reply src/main/resources/META-INF/persistence.xml]: I don't know if we even need to use `create` as `update` seems to do the same thing 🤔

**Comment by giplonis** [comment]
Nice! To move things quicker I also pushed some improvements while reviewing:
- Removed bidirecational relationships (they generally cause data consistency issues in application logic level)
- Removed unnecessary relationships (for example, Package does not need to know anything about Order)
- `OneToOne` is rarely needed and it's usually a sign of bad design. Same can be achieved using `ManyToOne`. More info [here](https://stackoverflow.com/questions/2452987/hibernate-why-use-many-to-one-to-represent-a-one-to-one).
- Added a couple of new entities cause I realized we'll need some extra data
- Rename `Package` to more sensible `PackageOption`
- Storing `float` in DB is suboptimal since it may introduce rounding errors, it's best to stick with `int` (x100)

**Review by giplonis — APPROVED** [review]

**Pull request merged**